### ISSUE TITLE
Editor - can't add 'Tabs' on first line

### ISSUE THREAD

**Issue by harryhobson** [spyder-ide/spyder#4038]
## Description

**What steps will reproduce the problem?**

1. open/create a python file
2. put cursor on line 1 column 1 (doesn't have to be first column if 'Tab Always Indent' is enabled)
3. press the 'Tab' key

**What is the expected output? What do you see instead?**

Indentation should be added (Tabs or spaces), instead the 'Internal Console' appears with a Trackback:
``` python
Traceback (most recent call last):
  File "/home/harry/anaconda2/lib/python2.7/site-packages/spyder/widgets/sourcecode/codeeditor.py", line 2805, in keyPressEvent
    self.indent_or_replace()
  File "/home/harry/anaconda2/lib/python2.7/site-packages/spyder/widgets/sourcecode/codeeditor.py", line 2111, in indent_or_replace
    self.indent()
  File "/home/harry/anaconda2/lib/python2.7/site-packages/spyder/widgets/sourcecode/codeeditor.py", line 2097, in indent
    if not self.fix_indent(forward=True):
  File "/home/harry/anaconda2/lib/python2.7/site-packages/spyder/widgets/sourcecode/codeeditor.py", line 1968, in fix_indent
    if prevtext.endswith(':') and self.is_python_like():
UnboundLocalError: local variable 'prevtext' referenced before assignment
```

**Please provide any additional information below**

Unsure if this is intentional or not, I've never needed to indent text on the first line of a python file.

## Version and main components

* Spyder Version: 3.1.0 
* Python Version: 2.7.13
* Qt Versions: 5.6.2, PyQt5 5.6 on Linux

## Dependencies
```
pyflakes >=0.5.0 :  1.5.0 (OK)
pep8 >=0.6       :  1.7.0 (OK)
pygments >=2.0   :  2.1.3 (OK)
qtconsole >=4.2.0:  4.2.1 (OK)
nbconvert >=4.0  :  4.2.0 (OK)
pandas >=0.13.1  :  0.19.2 (OK)
numpy >=1.7      :  1.11.3 (OK)
sphinx >=0.6.6   :  1.5.1 (OK)
rope >=0.9.4     :  0.9.4 (OK)
jedi >=0.8.1     :  0.9.0 (OK)
psutil >=0.3     :  5.0.1 (OK)
matplotlib >=1.0 :  1.5.3 (OK)
sympy >=0.7.3    :  1.0 (OK)
pylint >=0.25    :  1.6.4 (OK)

```

**Comment by harryhobson** [comment]
I've just realised this is related to #4021 which is addressed in PR #4029

**Comment by ccordoba12** [comment]
That's right. I'm closing this one as a duplicate of issue #4021.
